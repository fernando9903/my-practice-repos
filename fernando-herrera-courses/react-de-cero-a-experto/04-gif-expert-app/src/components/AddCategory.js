import React, { useState } from "react";

export const AddCategory = () => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (event) => {
        setInputValue(event.target.value);
    }

    const handleOnSubmit = (event) => {
        event.preventDefault();
        console.log('Submit hecho');
        setInputValue('');
    }

    return (
        <form onSubmit={handleOnSubmit}>
            <input
                type='text'
                placeholder='Type a new category that you want to add and press enter'
                onChange={handleInputChange}
            />
        </form>
    );
}