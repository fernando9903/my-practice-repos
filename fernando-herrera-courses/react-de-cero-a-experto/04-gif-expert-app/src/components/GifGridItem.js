import React from "react";
import PropTypes from 'prop-types';

export const GridItem = ({title, url}) => {

    return (
        <div className='card'>
            <img src={url} alt={title}/>
            <p>{title}</p>
        </div>
    )
}

GridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}