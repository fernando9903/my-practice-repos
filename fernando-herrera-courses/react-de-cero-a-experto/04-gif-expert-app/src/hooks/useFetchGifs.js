import { useState, useEffect } from "react"
import { getGifs } from "../helpers/getImages";

export const useFetchGifs = (category) => {
    const [state, setState] = useState({
        data: [],
        loading: true,
    })

    useEffect(() => {
         getGifs(category).then( images => {
            setTimeout( () => {
                setState({data: images, loading: false});
            }, 3000);
         })
    }, [category]);

    return state;

}