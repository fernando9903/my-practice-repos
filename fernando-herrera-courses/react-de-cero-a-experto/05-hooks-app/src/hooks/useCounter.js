import { useState } from "react"

export const useCounter = (initialState = 0) => {

    const [state, setState] = useState(initialState);

    const increment = (fact = 1) => {
        setState(state + fact)
    }

    const decrement = (fact = 1) => {
        setState(state - fact)
    }

    const reset = () => {
        setState(initialState);
    }

    return [
        state, 
        increment, 
        decrement, 
        reset
    ]

}