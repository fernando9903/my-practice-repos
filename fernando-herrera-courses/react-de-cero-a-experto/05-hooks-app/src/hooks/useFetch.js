import { useEffect, useRef, useState } from "react"

export const useFetch = ( url ) => {

    const isMounted = useRef(true);

    const [state, setState] = useState({
        loading: true,
        error: null,
        data: null,
    })

    useEffect(() => {
        fetch(url)
        .then( resp => resp.json())
        .then(data => {
            if (isMounted.current)
                setState({
                    loading: false,
                    error: null,
                    data,
                });
        })

        return () => {
            isMounted.current = false;
        }

    }, [url]);

    return state;

}