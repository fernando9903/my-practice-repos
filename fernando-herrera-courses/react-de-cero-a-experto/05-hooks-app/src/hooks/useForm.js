import { useState } from "react"

export const useForm = (initialState = {}) => {
    const [state, setState] = useState(initialState);

    const handleInputChange = (value = {}) => {
        setState({...state, value});
    }

    return [setState, handleInputChange];

}