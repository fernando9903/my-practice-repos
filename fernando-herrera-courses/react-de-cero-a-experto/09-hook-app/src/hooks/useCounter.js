import { useState } from "react";

export const useCounter = (initialValue = 0) => {

    const [value, setValue] = useState(initialValue);

    const increment = (fact = 1) => {
        setValue(value + fact);
    }

    const decrement = (fact = 1) => {
        console.log("Aqui toy")
        setValue(value - fact);
    }

    const reset = () => {
        setValue(initialValue);
    }

    return (
        increment,
        decrement,
        reset,
        value
    )

}