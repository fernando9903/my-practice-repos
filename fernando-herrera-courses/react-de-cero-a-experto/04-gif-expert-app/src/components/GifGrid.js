import React from "react";
import { useFetchGifs } from "../hooks/useFetchGifs";
import PropTypes from 'prop-types'

import { GridItem } from "./GifGridItem";

export const GifGrid = ({category}) => {

    const {loading, data:images} = useFetchGifs(category);

    return (
        <>
            <h3>{category}</h3>
            {loading ? 'Loading...' :
            <div className='card-grid animate__animated animate__fadeInTopLeft'>
                {images.map( image => 
                    <GridItem
                        key={image.id}
                        {...image}
                    />
                )}
            </div>}
        </>
    )
}   

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
}