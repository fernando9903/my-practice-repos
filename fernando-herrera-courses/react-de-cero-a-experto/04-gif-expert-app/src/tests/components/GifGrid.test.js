import React from "react";
import '@testing-library/jest-dom';
import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe('Test into <GifGrid> component', () => {

    test('The component shoul be render correctly', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true,
        });
    
        const category = 'Star wars';
        const wraper = shallow(
            <GifGrid
                key={category}
                category={category}
            />
        );
        expect(wraper).toMatchSnapshot();
    });

    test('Debe de mostrar alguna imagen cuando la peticion es lanzada', () => {
        const gifs = [{
            id: 'ABD',
            url: 'http://some/some.jpg',
            title: 'No importa el nombre'
        }]

        const category = 'some category';
        
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false,
        });
        const wraper = shallow(
            <GifGrid
                key={category}
                category={category}
            />
        );
        expect(wraper).toMatchSnapshot();
        expect(wraper.find('p').exists()).toBe(false);
        expect(wraper.find('GridItem').length).toBe(gifs.length);
    });

});