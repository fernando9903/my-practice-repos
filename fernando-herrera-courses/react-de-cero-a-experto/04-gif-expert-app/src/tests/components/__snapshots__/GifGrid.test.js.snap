// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test into <GifGrid> component Debe de mostrar alguna imagen cuando la peticion es lanzada 1`] = `
<Fragment>
  <h3>
    some category
  </h3>
  <div
    className="card-grid animate__animated animate__fadeInTopLeft"
  >
    <GridItem
      id="ABD"
      key="ABD"
      title="No importa el nombre"
      url="http://some/some.jpg"
    />
  </div>
</Fragment>
`;

exports[`Test into <GifGrid> component The component shoul be render correctly 1`] = `
<Fragment>
  <h3>
    Star wars
  </h3>
  Loading...
</Fragment>
`;
